src/backend/routers/deployment.py

```diff
@@ -167,21 +150,20 @@ async def delete_deployment(
     deployment = deployment_crud.get_deployment(session, deployment_id)
 
     if not deployment:
-        raise HTTPException(
-            status_code=404, detail=f"Deployment with ID: {deployment_id} not found."
-        )
+        raise DeploymentNotFoundError(deployment_id=deployment_id)
 
     deployment_crud.delete_deployment(session, deployment_id)
 
     return DeleteDeployment()
 
 
-@router.post("/{name}/set_env_vars", response_class=Response)
-async def set_env_vars(
-    name: str,
+@router.post("/{deployment_id}/update_config")
+async def update_config(
+    deployment_id: str,
+    session: DBSessionDep,
     env_vars: UpdateDeploymentEnv,
-    valid_env_vars=Depends(validate_env_vars),
-    ctx: Context = Depends(get_context),
+    valid_env_vars = Depends(validate_env_vars),
```

Comment:
We probably shouldn't do `env_var` validation as a `Depends`, and just do it in the body of this function

Reply:
Am good with that. Feel free to scope this in another ticket